src/doc/style-guide/src/expressions.md

```diff
@@ -536,6 +540,70 @@ if a_long_expression
 {
     ...
 }
+
+if let Some(a) = b
+    && another_long_expression
+    || a_third_long_expression
+{
+    ...
+}
+
+if let Some(relatively_long_thing)
+    = a_long_expression
+    && another_long_expression
+    || a_third_long_expression
+{
+    ...
+}
+
+if some_expr
+    && another_long_expression
+    && let Some(relatively_long_thing)
+        = a_long_expression
+    || a_third_long_expression
+{
+    ...
+}
+```
+
+A let-chain control line is allowed to be formatted on a single line provided
+it only consists of two clauses, separated by `&&`, with the first being an
+`ident` (which can optionally be preceded by any number of unary prefix operators)
```

Comment:
I think we should change this to "ident or literal", so that we don't break after `false &&` or `true &&`.